Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts

```diff
@@ -27,24 +31,24 @@ class Credential {
 function getRequest(url: string, cred: Credential, strictSSL: boolean): Q.Promise<any> {
     const defer = Q.defer<any>();
 
-    request.get({url: url, strictSSL: strictSSL}, (err, res, body) => {
-                if (res && body && res.statusCode === 200)  {
-                    defer.resolve(JSON.parse(body));
-                } else {
-                    if (res && res.statusCode) {
-                        tl.debug(tl.loc('ServerCallErrorCode', res.statusCode));
-                    }
-                    if (body) {
-                        tl.debug(body);
-                    }
-                    defer.reject(new Error(tl.loc('ServerCallFailed')));
-                }
-            })
-            .auth(cred.mUsername, cred.mPassword, true)
-            .on('error', (err) => {
-                //TODO: Do we even need an 'error' handler here if we're just re-throwing?
-                defer.reject(new Error(err.message));
-            });
+    request.get({ url: url, strictSSL: strictSSL }, (err, res, body) => {
```

Comment:
can we use typed-rest-client here? I think 'request' is a very big dependency

Reply:
I'll let Omesh know this and make a decision.

Reply:
We use request in a several other tasks. It has a problem with agent proxy settings not flowing as well. Where ever it makes sense, we should move to typed-rest-client. Looks like there was a dependency on request here before as well, so it is not adding anything to the task. But will be good to revisit later.